### PR TITLE
Add support for XML marshalling/unmarshalling

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

File: src/generator/protocol/operations.ts [hunk 1]
```diff
@@ -353,6 +353,29 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
...
+        if (isArraySchema(bodyParam!.schema)) {
+          fieldName = pascalCase(bodyParam!.language.go!.name);
+          let tag = bodyParam!.schema.elementType.language.go!.name;
+          if (bodyParam!.schema.elementType.serialization?.xml?.name) {
+            tag = bodyParam!.schema.elementType.serialization.xml.name;
+          }
+          text += `\t\t${fieldName} *${bodyParam!.schema.language.go!.name} \`xml:"${tag}"\`\n`;
+        } else {
+          // embed as anonymous field
+          text += `\t\t*${bodyParam!.schema.language.go!.name}\n`;
```
Comment (catalinaperalta) [inline src/generator/protocol/operations.ts]: Just trying to understand this a little more, what cases do we embed this anonymously?

Comment (jhendrixMSFT) [reply src/generator/protocol/operations.ts]: If it's an object then it's embedded anonymously (there are only two cases, arrays or objects).

File: test/autorest/generated/urlgroup/internal/urlgroup/paths.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (PathsOperations) ByteNullCreateRequest(u url.URL, bytePath []byte) (*azcor
 
 // ByteNullHandleResponse handles the ByteNull response.
 func (PathsOperations) ByteNullHandleResponse(resp *azcore.Response) (*PathsByteNullResponse, error) {
-	if !resp.HasStatusCode(http.StatusOK) {
+	if !resp.HasStatusCode(http.StatusBadRequest) {
```
Comment (catalinaperalta) [inline test/autorest/generated/urlgroup/internal/urlgroup/paths.go]: We're expecting null scenarios to return a BadRequest status?

File: src/generator/protocol/operations.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/generator/protocol/operations.ts]: Ah ok, good to know. Thanks!

File: test/autorest/generated/urlgroup/internal/urlgroup/paths.go (hunk 2 above)
Comment (catalinaperalta) [reply test/autorest/generated/urlgroup/internal/urlgroup/paths.go]: I know this comes from the swagger, but I'm wondering what changed. Also, shouldnt it be without the ! for bad requests? That would be an error wouldnt it?

Comment (jhendrixMSFT) [reply test/autorest/generated/urlgroup/internal/urlgroup/paths.go]: The way it's worded in the test, the operation returns invalid data that should cause unmarshalling to fail so the API call returns an error.  So if you actually get back the 400 and no error something went wrong elsewhere (it's kinda weird).

Comment (jhendrixMSFT) [reply test/autorest/generated/urlgroup/internal/urlgroup/paths.go]: As for what changed, this was fixed via `formatStatusCodes()`.  Before that we had a hard-coded `http.StatusOK`.

Comment (catalinaperalta) [reply test/autorest/generated/urlgroup/internal/urlgroup/paths.go]: Hmmm it is very strange. But I guess this is the expected output then. Doesnt sound like real world scenarios should work like this though.

**Pull request merged**